docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+        """
+        if not hasattr(self, "_package_manager"):
+            raise AttributeError(  # pragma: no cover
+                "Attribute 'package_manager' not found."
+            )
+        return getattr(self, "_package_manager")
+
+
+class AptGetCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for debian-based apt-get systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-dev-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-dev-$CUDA_PKG_VERSION",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(AptGetCUDADockerfileGen, self).__init__(url_reader=url_reader)
+        self._package_manager: PackageManager = get_package_manager("apt-get")
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        filename = f"cuda-keyring_1.0-1_all.{self.package_manager.file_type}"
+        read_section = self.url_reader.generate_read_command(
+            target=f"{cuda_repo_name}" + filename,
+            output_file=filename,
+        )
+        package_command_section = self.package_manager.generate_package_command(
+            target=filename,
+        )
+        return f"RUN {read_section} \\\n && {package_command_section}"
+
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        targets = build_targets
+        install_section = self.package_manager.generate_install_command(
+            targets=targets, clean=True
+        )
+        install_line = (
+            f"RUN {self.package_manager.name} update \\\n && {install_section}"
+        )
+        return install_line
+
+
+class YumCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for rpm-based yum systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-devel-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-devel-$CUDA_PKG_VERSION",
+        "rpm-build",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(YumCUDADockerfileGen, self).__init__(url_reader=url_reader)
+        self._package_manager: PackageManager = get_package_manager("yum")
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        cuda_repo = f"{cuda_repo_name}cuda-{repo_ver}.repo"
+        return f"RUN yum-config-manager --add-repo {cuda_repo} \\\n " "&& yum clean all"
+
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        targets = build_targets
+        install_line = self.package_manager.generate_install_command(
+            targets=targets, clean=True
```

Comment:
No need to rename `build_targets` to `targets` in the function body.

Maybe we should just change the name of the function parameter to `targets` instead of `build_targets` if you like that name better?

```suggestion
        install_line = self.package_manager.generate_install_command(
            targets=build_targets, clean=True
```